src/voice.ts

```diff
@@ -73,51 +98,74 @@ export class Voice extends Element {
   }
 
   // Get the total ticks in the voice
-  getTotalTicks() { return this.totalTicks; }
+  getTotalTicks(): Fraction {
+    return this.totalTicks;
+  }
 
   // Get the total ticks used in the voice by all the tickables
-  getTicksUsed() { return this.ticksUsed; }
+  getTicksUsed(): Fraction {
+    return this.ticksUsed;
+  }
 
   // Get the largest width of all the tickables
-  getLargestTickWidth() { return this.largestTickWidth; }
+  getLargestTickWidth(): number {
+    return this.largestTickWidth;
+  }
 
   // Get the tick count for the shortest tickable
-  getSmallestTickCount() { return this.smallestTickCount; }
+  getSmallestTickCount(): Fraction {
+    return this.smallestTickCount;
+  }
 
   // Get the tickables in the voice
-  getTickables() { return this.tickables; }
+  getTickables(): Note[] {
+    return this.tickables;
+  }
 
   // Get/set the voice mode, use a value from `Voice.Mode`
-  getMode() { return this.mode; }
-  setMode(mode) { this.mode = mode; return this; }
+  getMode(): number {
+    return this.mode;
+  }
+
+  setMode(mode: number): this {
+    this.mode = mode;
+    return this;
+  }
 
   // Get the resolution multiplier for the voice
-  getResolutionMultiplier() { return this.resolutionMultiplier; }
+  getResolutionMultiplier(): number {
+    return this.resolutionMultiplier;
+  }
 
   // Get the actual tick resolution for the voice
-  getActualResolution() { return this.resolutionMultiplier * this.time.resolution; }
+  getActualResolution(): number {
+    if (typeof (this.time) === 'string') {
+      throw new RuntimeError('VoiceError', 'Time is of type string');
+    }
+    return this.resolutionMultiplier * this.time.resolution;
+  }
 
   // Set the voice's stave
-  setStave(stave) {
+  setStave(stave: Stave): this {
     this.stave = stave;
     this.boundingBox = null; // Reset bounding box so we can reformat
     return this;
   }
 
   // Get the bounding box for the voice
-  getBoundingBox() {
+  getBoundingBox(): BoundingBox {
     let stave;
     let boundingBox;
     let bb;
     let i;
 
     if (!this.boundingBox) {
-      if (!this.stave) throw new Vex.RERR('NoStave', "Can't get bounding box without stave.");
+      if (!this.stave) throw new RuntimeError('NoStave', "Can't get bounding box without stave.");
       stave = this.stave;
       boundingBox = null;
 
       for (i = 0; i < this.tickables.length; ++i) {
-        this.tickables[i].setStave(stave);
+        (this.tickables[i] as Note).setStave(stave);
```

Comment:
this.tickables is of type Note already. I believe `as Note` is not required. There are several instances in this file.

Reply:
That can be, I will check. Thanks